test/testdata/resolver/basic_constant_out_of_order.rb

```diff
@@ -0,0 +1,46 @@
+# enable-out-of-order-reference-checks: true
+# typed: true
+
+module Foo
+  A = X 
+    # ^ error: `Foo::X` referenced before it is defined
+
+  def self.foo(arg:)
+    X # this is ok
+  end
+
+  def self.bar(&blk)
+    X # this is ok
+  end
+
+  foo arg: ->{ X } # this is ok
+
+  bar do
+    X # this is ok
+  end
+
+  class X; end
+
+  B = X # this is ok
+
+  class Bar
+    Foo::Y
+  # ^^^^^^ error: `Foo::Y` referenced before it is defined
```

Comment:
```suggestion
    p(Foo::Y)
    # ^^^^^^ error: `Foo::Y` referenced before it is defined
```

this is another minor thing but

- using `p` makes it look like the constant isn't unused (ALE in vim by default runs `ruby -w` which prints `warning: possibly useless use of a constant in void context`)
- using `p(` also gives us some extra characters to allow the `^^^` to line up nicely

Reply:
👍